### PR TITLE
added possibility to use urls with query string

### PR DESCRIPTION
**Pull request by ausi**

### CODE REVIEW

File: Source/History.js [hunk 1]
```diff
@@ -42,24 +48,28 @@ this.History = new new Class({
 			this.timer = this.check.periodical(200, this);
 	},
 
+	getUrlPart: getUrlPart,
```
Comment (cpojer) [inline Source/History.js]: Why does this need to be public?

File: Source/History.js [hunk 2]
```diff
@@ -20,7 +20,13 @@ provides: History
 
 var events = Element.NativeEvents,
 	location = window.location,
-	base = location.pathname,
+	getUrlPart = function(url){
+		if(url.substr(0,8) == 'https://' || url.substr(0,7) == 'http://'){
+			url = '/'+url.split('/').slice(3).join('/');
+		}
```
Comment (cpojer) [inline Source/History.js]: would you mind updating the code style here to match the project's code style?

Something like this:

if (url.match(/https?:\/\//) url = '/' + url.split('/').slice(3).join('/');

Also, why not just name the method cleanURL(), getURL() or something simple like that?

**Comment by cpojer** [comment]
Awesome! I've been meaning to do this but never got around to it. Would you mind addressing my inline comments?

File: Source/History.js (hunk 1 above)
Comment (ausi) [inline Source/History.js]: because it is used in `History.handleInitialState.js`, is there a better way to do this?

Comment (cpojer) [inline Source/History.js]: Oh wow, totally missed that. I guess it is fine then.

File: Source/History.js (hunk 2 above)
Comment (ausi) [inline Source/History.js]: done

**Comment by ausi** [comment]
I've changed the code according to your inline comments.
Thanks for this great MooTools Class!

**Comment by cpojer** [comment]
awesome! Thank you.

**Comment by cpojer** [comment]
Is there a reason you are removing the #hash part of the URL? I'd like to keep that.

**Comment by ausi** [comment]
I thought not removing the #hash would cause problems in browsers not supporting push state, but i didn't tested it.

**Pull request merged**